tools/build-bpf.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: LGPL-2.1+
+# Copyright (c) 2020 Facebook
+
+import argparse
+import logging
+import os.path
+import shutil
+import subprocess
+import sys
+import tempfile
+
+hexdump_src_template='''
+/* SPDX-License-Identifier: LGPL-2.1+ */
+/* Autogenerated source header with byte array for %src_c% BPF program.
+ * DO NOT MODIFY by hand, will be overriden by meson build rule.
+ */
+#pragma once
+
+const unsigned char %buffer_name%[] = {
+%hexdump%
+};
+'''
+
+def clang_compile(clang_exec, clang_flags, src_c, src_h, out_file, target_arch,
+        target_triplet=None):
+    clang_args = [clang_exec]
+    clang_args.extend(clang_flags)
+    clang_args.extend([target_arch])
+
+    clang_args.extend([
+        '-I.'])
+
+    if target_triplet:
+        clang_args.extend([
+            '-isystem',
+            '/usr/include/{}'.format(target_triplet)])
+
+    clang_args.extend([
+        '-idirafter',
+        '/usr/local/include',
+        '-idirafter',
+        '/usr/include'])
+
+    clang_args.extend([src_c])
+    clang_args.extend(['-o', out_file])
+
+    logging.debug('Generating LLVM bitcode *.bc:')
+    logging.debug('{}'.format(' '.join(clang_args)))
+    subprocess.check_call(clang_args)
+
+
+def llc_compile(llc_exec, llc_flags, in_file, out_file):
+    llc_args = [llc_exec]
+    llc_args.extend(llc_flags)
+    llc_args.extend([out_file, in_file])
+
+    logging.debug('Compiling BPF object file:')
+    logging.debug('{}'.format(' '.join(llc_args)))
+    subprocess.check_call(llc_args)
+
+
+def hexdump(in_file, indent):
+    ret = ''
+    indent_prefix = ' ' * indent
+    with open(in_file, 'rb') as in_fd:
+        while True:
+            octets = in_fd.read(16)
+            if not octets:
+                break
+            # Represent octet as hexadecimal number in range 0x00-0xff,
+            # join numbers into a line.
+            line = ', '.join(['{:#04x}'.format(o) for o in octets])
+            ret += '{}{}'.format(indent_prefix, line)
+            indent_prefix = ',\n{}'.format(' ' * indent)
+    return ret
+
+
+def get_default_hexdump_buffer_name(hexdump_file):
+    return os.path.splitext(os.path.basename(hexdump_file))[0].replace('-',
+            '_') + '_buffer'
+
+
+def get_arch_clang_flag(arch):
+    return '-D__{}__'.format(arch)
+
+
+def get_target_triplet():
+    gcc_exec = None
+    try:
+        gcc_exec = shutil.which('gcc')
+    except shutil.Error as e:
+        logging.error('Failed to get which gcc: {}'.format(e))
+        return None
+
+    try:
+        return subprocess.check_output([gcc_exec, '-dumpmachine'],
+                text=True).strip()
```

Comment:
`text` was introduced in Python 3.7, what about using `universal_newlines` to keep compatibility with older versions? Otherwise to update the required Python version?